src/Pos/Explorer/DB.hs

```diff
@@ -77,17 +99,38 @@ putGenesisBalances genesisUtxo =
 data ExplorerOp
     = AddTxExtra !TxId !TxExtra
     | DelTxExtra !TxId
+
+    | PutPageBlocks !Page ![HeaderHash]
+    | DelPageBlocks !Page
+
+    | PutEpochBlocks !Epoch ![HeaderHash]
+    | DelEpochBlocks !Epoch
```

Comment:
Btw, `Del` operations are not used. There are two options:
1. In rollback you can check whether the list you put is empty (it might be empty sometimes) and use `Del` operation if it is.
2. You can never delete, because even if rollback happens and we put empty list, we know that it will be refilled very soon, so it's negligible overhead to store empty lists. In this case I propose to remove these operations altogether.